### ISSUE TITLE
[Issue]: InputValidationError returns text/plain but should return application/json

### ISSUE THREAD

**Issue by anothermh** [wundergraph/wundergraph#1319]
### Description

When using WunderGraph I found that input validation errors are returned to the caller with `Content-Type: text/plain; charset=utf-8`. I expect that the errors should be returned as `application/json` because the text that is returned *is* JSON, just with the wrong `Content-Type` header.

Here's an example response showing the issue:

```
HTTP/1.1 400 Bad Request
Vary: Origin
Date: Tue, 17 Oct 2023 21:05:51 GMT
Content-Length: 247
Content-Type: text/plain; charset=utf-8
Connection: close

{"code":"InputValidationError","message":"Bad Request: Invalid input","input":{},"errors":[{"propertyPath":"/","invalidValue":{},"message":"\"foo\" value is required"},{"propertyPath":"/","invalidValue":{},"message":"\"bar\" value is required"}]}
```

### Reproduction

Runtime: 20.4.0
NPM: 9.7.2

From the [Quickstart](https://docs.wundergraph.com/docs/getting-started/quickstart#creating-a-new-wunder-graph-project):

```shell
npx create-wundergraph-app my-project --example simple
cd my-project
npm install
```

Then create a new TypeScript operation at `.wundergraph/operations/foo.ts`:

```typescript
import { z } from '@wundergraph/sdk/operations'
import { createOperation } from '../generated/wundergraph.factory'

export default createOperation.query({
  input: z.object({
    foo: z.string(),
    bar: z.string(),
  }),
  handler: async (input: any): Promise<any> => {
    console.log('Input value is:', JSON.stringify(input))

    return {
      baz: 'foobarbaz',
    }
  },
})
```

Then `npm build && npm start`. Then make a request to the endpoint:

```
GET /operations/foo HTTP/1.1
Content-Type: application/json
Host: localhost:9991
Connection: close
```

And receive the response:

```
HTTP/1.1 400 Bad Request
Vary: Origin
Date: Tue, 17 Oct 2023 21:05:51 GMT
Content-Length: 247
Content-Type: text/plain; charset=utf-8
Connection: close

{"code":"InputValidationError","message":"Bad Request: Invalid input","input":{},"errors":[{"propertyPath":"/","invalidValue":{},"message":"\"foo\" value is required"},{"propertyPath":"/","invalidValue":{},"message":"\"bar\" value is required"}]}
```

My expectation is that the JSON response returned here should have the Content-Type header set to `application/json` because the response is JSON:

```json
{
  "code": "InputValidationError",
  "message": "Bad Request: Invalid input",
  "input": {},
  "errors": [
    {
      "propertyPath": "/",
      "invalidValue": {},
      "message": "\"foo\" value is required"
    },
    {
      "propertyPath": "/",
      "invalidValue": {},
      "message": "\"bar\" value is required"
    }
  ]
}
```

This error is handled at https://github.com/wundergraph/wundergraph/blob/7b2d09797e28548a9f7afdfac8731db719a6ad93/pkg/inputvariables/validator.go#L41-L43

It appears that the error is returned to the caller at https://github.com/wundergraph/wundergraph/blob/7b2d09797e28548a9f7afdfac8731db719a6ad93/pkg/apihandler/apihandler.go#L435-L438

**Comment by Pagebakers** [comment]
Thanks for the elaborate report, this seems to be incorrect indeed. Have you also looked into a possible fix?

**Comment by anothermh** [comment]
I've tried a number of different fixes where I use `w.Header().Set("Content-Type", "application/json")` prior to the error response (or any response) being sent but it never seems to stick. For example, I thought this block might be where to set the header but that wasn't it: https://github.com/wundergraph/wundergraph/blob/38c73f9aed26cc1a21abd769ca778f8a3576da2c/pkg/apihandler/errorhandler.go#L101-L109

`api.go` and `apihandler.go` and `errorhandler.go` and `internalapihandler.go` combined are over 3k LOC so there's a lot of ground to cover in terms of understanding how it comes together. I don't frequently use Golang so it's possible that I'm missing an obvious fix. 

Anybody is welcome to make a PR for this if they can find the right way to set the header in the response for validation errors. Unfortunately I've exhausted the time I can spend on it for now.
